command/agent/agent_auto_auth_self_heal_test.go

```diff
@@ -154,33 +126,33 @@ func TestAutoAuthSelfHealing_TokenFileAuth_SinkOutput(t *testing.T) {
 		ExitAfterAuth: false,
 	}
 
+	pathTemplateOutput := makeTempFile(t, "template-output", "")
+	originalTemplateFileInfo, err := os.Stat(pathSinkFile)
+	require.NoError(t, err)
 	templateTest := &ctconfig.TemplateConfig{
-		Contents:    pointerutil.StringPtr(lookupSelfTemplateContents),
-		Destination: pointerutil.StringPtr(pathLookupSelf),
+		Contents:    pointerutil.StringPtr(`{{ with secret "auth/token/lookup-self" }}{{ .Data.id }}{{ end }}`),
+		Destination: pointerutil.StringPtr(pathTemplateOutput),
 	}
 	templatesToRender := []*ctconfig.TemplateConfig{templateTest}
 
-	var server *template.Server
-	server = template.NewServer(sc)
+	server := template.NewServer(sc)
 	go func() {
 		errCh <- server.Run(ctx, ah.TemplateTokenCh, templatesToRender, ah.AuthInProgress, ah.InvalidToken)
 	}()
 
-	// Trigger template render (mark the time as being earlier, based on the render interval)
-	preTriggerTime := time.Now().Add(-secretRenderInterval)
+	// Send token to auth channel
 	ah.TemplateTokenCh <- token
-	fileInfo, err := waitForFiles(t, pathTokenFile, preTriggerTime)
```

Comment:
This was the root cause - the pre-trigger time might be before the sink we created earlier. Instead, we're waiting for the sink to be updated past the sink's time